src/coreclr/utilcode/executableallocator.cpp

```diff
@@ -260,7 +261,7 @@ HRESULT ExecutableAllocator::StaticInitialize(FatalErrorHandler fatalErrorHandle
 #endif
 
     g_fatalErrorHandler = fatalErrorHandler;
-    g_isWXorXEnabled = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableWriteXorExecute) != 0;
+    g_isWXorXEnabled = Configuration::GetKnobBooleanValue(W("System.Runtime.EnableWriteXorExecute"), CLRConfig::EXTERNAL_EnableWriteXorExecute, TRUE);
```

Comment:
I actually find the three argument version of the function a bit confusing. The thing is that there are two places where the default is defined - once at the CLRConfig::EXTERNAL_EnableWriteXorExecute definition site and once here. This version basically ignores the default at the CLRConfig::EXTERNAL_EnableWriteXorExecute definition site.
So, I'd prefer the way you had it before - without the last argument here. And I believe it was correct as it was, it didn't need any comparison to 0 as it returns `bool`.
The previous version was using the default defined at the CLRConfig::EXTERNAL_EnableWriteXorExecute definition site when neither the env var nor the runtimeConfig setting was present.

Reply:
Oh. That explains why both the code and my tests showed that all works as intended.